draft-ietf-httpbis-message-signatures.md

```diff
@@ -1470,39 +1499,47 @@ Content-Digest: sha-512=:WZDPaVn/7XgHaAy8pmojAkGWoRx2UFChF41A2svX+T\
 Signature-Input: sig1=("@method" "@authority" "@path" \
   "content-digest" "content-type" "content-length")\
   ;created=1618884475;keyid="test-key-ecc-p256"
-Signature: sig1=:hNojB+wWw4A7SYF3qK1S01Y4UP5i2JZFYa2WOlMB4Np5iWmJSO\
-  0bDe2hrYRbcIWqVAFjuuCBRsB7lYQJkzbb6g==:
+Signature: sig1=:e7vqNlt7NV8mJXXja35MQRW/puA3JOHCsKiAG5U7lnpwaqMqn6\
+  83Cna7hyXF4rnpyACdJcOWFB3FSPVymw0fzA==:
 
 {"hello": "world"}
 ~~~
 
 While the proxy is in a position to validate the client's signature, the changes the proxy makes to the message will invalidate the existing signature when the message is seen by the origin server. While it is possible for the origin server to have additional information in its signature context to account for the change in authority, this practice requires additional configuration and extra care (see further discussion in {{security-context-multiple-signatures}}). To counter this, the proxy adds its own signature over the new message before passing it along. The proxy includes the new `@authority` derived component and the Forwarded header, which it added to the message.
-The proxy's signature also includes the client's signature value from the original message in its covered components, as a dictionary field member under the label `sig1`. Note that since the client's signature already covers the client's Signature-Input value for `sig1`, this value is transitively covered by the proxy's signature and need not be added explicitly. While the origin server may not be able to directly verify this original signature, it can verify that the proxy has vouched for the signature's validity.
+The proxy's signature also includes the client's signature value and signature input from the original message in its covered components, as dictionary field members under the label `sig1`. The proxy can also sign any unchanged components from the original message, such as the `@method`, `@path`, and `@query` in this example, even if those components were covered by the original signature. See the discussion in {{security-sign-signature}} for more information about signing signature values.
```

Comment:
Does this need to say "client's signature value" given the attacks that Dennis describes?

Reply:
@martinthomson not sure I understand the comment -- are you saying the signature value shouldn't be included in the proxy's signature at all, or that where it says "original signature" it should instead read "the client's signature"? If the former, the attack surface is a little bit different here because the proxy is a trusted party that's making a different kind of statement than the request-response-binding non-repudiation. This is mitigated by covering additional portions of the message, including the signature input line and any items that didn't change, along with items that did change at the proxy.

Reply:
I guess the problem here is that the specific example is hard to follow.  Is it fair to say that the proxy has two goals: 1. it wants to cover the request that arrives at the server and 2. it also wants to transitively attest to what the client requested of it (maybe because the server can also validate that, but this is not entirely clear from context).

The former does not require that the signature or signature input from the client needs to be covered at all.  The latter is more interesting and requires that the proxy not change anything that the client signed, it can only add things that it additionally signs.  But in that case, the proxy can simply sign the same stuff the client did (and anything extra).  The only thing that is left is signing the fact that it knows that the client signed the message or signing to signal that it validated the client's signature (what signing a signature means here between those two, I don't know, maybe the difference is encoded in the difference between signing `signature-input` and signing `signature`).

Either way, the emphasis on including the signature is probably not really advancing a particular goal and should be de-emphasized.